src/piece/Pawn.hpp

```diff
@@ -8,7 +8,8 @@ namespace chesspp
 {
     namespace piece
     {
-        class Pawn : public virtual Piece
+        class Pawn
```

Comment:
Yeah I say modify it or add a new entry for inheritance. It wasn't obvious to me that it was meant for both situations, which means it'll catch someone else also.